streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -997,6 +1064,72 @@ private boolean close(final long timeoutMs) {
         }
     }
 
+    private void closeToError() {
+        if (!setState(State.ERROR)) {
+            // if transition failed, it means it was either in PENDING_SHUTDOWN
+            // or NOT_RUNNING already; just check that all threads have been stopped
+            log.info("Can not close to error from state " + state());
```

Comment:
```suggestion
            log.info("Can not transition to error from state " + state());
```

Didn't follow the prior message. Is this what you meant?

Reply:
That works